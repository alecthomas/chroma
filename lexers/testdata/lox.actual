// Your first Lox program!
print "Hello, world!";

true;  // Not false.
false; // Not *not* false.

1234;  // An integer.
12.34; // A decimal number.

"I am a string";
"";    // The empty string.
"123"; // This is a string, not a number.

nil;

add + me;
subtract - me;
multiply * me;
divide / me;

-negateMe;

less < than;
lessThan <= orEqual;
greater > than;
greaterThan >= orEqual;

1 == 2;         // false.
"cat" != "dog"; // true.

314 == "pi"; // false.

123 == "123"; // false.

!true;  // false.
!false; // true.

true and false; // false.
true and true;  // true.

false or false; // false.
true or false;  // true.

{
  print "One statement.";
  print "Two statements.";
}

var imAVariable = "here is my value";
var iAmNil;

var breakfast = "bagels";
print breakfast; // "bagels".
breakfast = "beignets";
print breakfast; // "beignets".

if (condition) {
  print "yes";
} else {
  print "no";
}

var a = 1;
while (a < 10) {
  print a;
  a = a + 1;
}

for (var a = 1; a < 10; a = a + 1) {
  print a;
}

makeBreakfast(bacon, eggs, toast);

makeBreakfast();

fun addPair(a, b) {
  return a + b;
}
fun identity(a) {
  return a;
}
print identity(addPair)(1, 2); // Prints "3".

fun returnFunction() {
  var outside = "outside";
  fun inner() {
    print outside;
  }
  return inner;
}
var fn = returnFunction();
fn();

class Breakfast {
  cook() {
    print "Eggs a-fryin'!";
  }
  serve(who) {
    print "Enjoy your breakfast, " + who + ".";
  }
}

// Store it in variables.
var someVariable = Breakfast;

// Pass it to functions.
someFunction(Breakfast);

breakfast.meat = "sausage";
breakfast.bread = "sourdough";

class Breakfast {
  init(meat, bread) {
    this.meat = meat;
    this.bread = bread;
  }
  // ...
}
var baconAndToast = Breakfast("bacon", "toast");
baconAndToast.serve("Dear Reader");
// "Enjoy your bacon and toast, Dear Reader."

class Brunch < Breakfast {
  init(meat, bread, drink) {
    super.init(meat, bread);
    this.drink = drink;
  }
  drink() {
    print "How about a Bloody Mary?";
  }
}
