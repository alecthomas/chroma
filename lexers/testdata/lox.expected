[
  {"type":"CommentSingle","value":"// Your first Lox program!\n"},
  {"type":"Keyword","value":"print"},
  {"type":"Text","value":" "},
  {"type":"LiteralStringDouble","value":"\"Hello, world!\""},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n\n"},
  {"type":"KeywordConstant","value":"true"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"  "},
  {"type":"CommentSingle","value":"// Not false.\n"},
  {"type":"KeywordConstant","value":"false"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":" "},
  {"type":"CommentSingle","value":"// Not *not* false.\n"},
  {"type":"Text","value":"\n"},
  {"type":"LiteralNumberInteger","value":"1234"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"  "},
  {"type":"CommentSingle","value":"// An integer.\n"},
  {"type":"LiteralNumberFloat","value":"12.34"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":" "},
  {"type":"CommentSingle","value":"// A decimal number.\n"},
  {"type":"Text","value":"\n"},
  {"type":"LiteralStringDouble","value":"\"I am a string\""},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n"},
  {"type":"LiteralStringDouble","value":"\"\""},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"    "},
  {"type":"CommentSingle","value":"// The empty string.\n"},
  {"type":"LiteralStringDouble","value":"\"123\""},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":" "},
  {"type":"CommentSingle","value":"// This is a string, not a number.\n"},
  {"type":"Text","value":"\n"},
  {"type":"KeywordConstant","value":"nil"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n\n"},
  {"type":"NameVariable","value":"add"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"+"},
  {"type":"Text","value":" "},
  {"type":"NameVariable","value":"me"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n"},
  {"type":"NameVariable","value":"subtract"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"-"},
  {"type":"Text","value":" "},
  {"type":"NameVariable","value":"me"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n"},
  {"type":"NameVariable","value":"multiply"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"*"},
  {"type":"Text","value":" "},
  {"type":"NameVariable","value":"me"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n"},
  {"type":"NameVariable","value":"divide"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"/"},
  {"type":"Text","value":" "},
  {"type":"NameVariable","value":"me"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n\n"},
  {"type":"Operator","value":"-"},
  {"type":"NameVariable","value":"negateMe"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n\n"},
  {"type":"NameVariable","value":"less"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"\u003c"},
  {"type":"Text","value":" "},
  {"type":"NameVariable","value":"than"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n"},
  {"type":"NameVariable","value":"lessThan"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"\u003c="},
  {"type":"Text","value":" "},
  {"type":"NameVariable","value":"orEqual"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n"},
  {"type":"NameVariable","value":"greater"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"\u003e"},
  {"type":"Text","value":" "},
  {"type":"NameVariable","value":"than"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n"},
  {"type":"NameVariable","value":"greaterThan"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"\u003e="},
  {"type":"Text","value":" "},
  {"type":"NameVariable","value":"orEqual"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n\n"},
  {"type":"LiteralNumberInteger","value":"1"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"=="},
  {"type":"Text","value":" "},
  {"type":"LiteralNumberInteger","value":"2"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"         "},
  {"type":"CommentSingle","value":"// false.\n"},
  {"type":"LiteralStringDouble","value":"\"cat\""},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"!="},
  {"type":"Text","value":" "},
  {"type":"LiteralStringDouble","value":"\"dog\""},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":" "},
  {"type":"CommentSingle","value":"// true.\n"},
  {"type":"Text","value":"\n"},
  {"type":"LiteralNumberInteger","value":"314"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"=="},
  {"type":"Text","value":" "},
  {"type":"LiteralStringDouble","value":"\"pi\""},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":" "},
  {"type":"CommentSingle","value":"// false.\n"},
  {"type":"Text","value":"\n"},
  {"type":"LiteralNumberInteger","value":"123"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"=="},
  {"type":"Text","value":" "},
  {"type":"LiteralStringDouble","value":"\"123\""},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":" "},
  {"type":"CommentSingle","value":"// false.\n"},
  {"type":"Text","value":"\n"},
  {"type":"Operator","value":"!"},
  {"type":"KeywordConstant","value":"true"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"  "},
  {"type":"CommentSingle","value":"// false.\n"},
  {"type":"Operator","value":"!"},
  {"type":"KeywordConstant","value":"false"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":" "},
  {"type":"CommentSingle","value":"// true.\n"},
  {"type":"Text","value":"\n"},
  {"type":"KeywordConstant","value":"true"},
  {"type":"Text","value":" "},
  {"type":"Keyword","value":"and"},
  {"type":"Text","value":" "},
  {"type":"KeywordConstant","value":"false"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":" "},
  {"type":"CommentSingle","value":"// false.\n"},
  {"type":"KeywordConstant","value":"true"},
  {"type":"Text","value":" "},
  {"type":"Keyword","value":"and"},
  {"type":"Text","value":" "},
  {"type":"KeywordConstant","value":"true"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"  "},
  {"type":"CommentSingle","value":"// true.\n"},
  {"type":"Text","value":"\n"},
  {"type":"KeywordConstant","value":"false"},
  {"type":"Text","value":" "},
  {"type":"Keyword","value":"or"},
  {"type":"Text","value":" "},
  {"type":"KeywordConstant","value":"false"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":" "},
  {"type":"CommentSingle","value":"// false.\n"},
  {"type":"KeywordConstant","value":"true"},
  {"type":"Text","value":" "},
  {"type":"Keyword","value":"or"},
  {"type":"Text","value":" "},
  {"type":"KeywordConstant","value":"false"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"  "},
  {"type":"CommentSingle","value":"// true.\n"},
  {"type":"Text","value":"\n"},
  {"type":"Punctuation","value":"{"},
  {"type":"Text","value":"\n  "},
  {"type":"Keyword","value":"print"},
  {"type":"Text","value":" "},
  {"type":"LiteralStringDouble","value":"\"One statement.\""},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n  "},
  {"type":"Keyword","value":"print"},
  {"type":"Text","value":" "},
  {"type":"LiteralStringDouble","value":"\"Two statements.\""},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n"},
  {"type":"Punctuation","value":"}"},
  {"type":"Text","value":"\n\n"},
  {"type":"KeywordDeclaration","value":"var "},
  {"type":"NameVariable","value":"imAVariable"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"="},
  {"type":"Text","value":" "},
  {"type":"LiteralStringDouble","value":"\"here is my value\""},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n"},
  {"type":"KeywordDeclaration","value":"var "},
  {"type":"NameVariable","value":"iAmNil"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n\n"},
  {"type":"KeywordDeclaration","value":"var "},
  {"type":"NameVariable","value":"breakfast"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"="},
  {"type":"Text","value":" "},
  {"type":"LiteralStringDouble","value":"\"bagels\""},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n"},
  {"type":"Keyword","value":"print"},
  {"type":"Text","value":" "},
  {"type":"NameVariable","value":"breakfast"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":" "},
  {"type":"CommentSingle","value":"// \"bagels\".\n"},
  {"type":"NameVariable","value":"breakfast"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"="},
  {"type":"Text","value":" "},
  {"type":"LiteralStringDouble","value":"\"beignets\""},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n"},
  {"type":"Keyword","value":"print"},
  {"type":"Text","value":" "},
  {"type":"NameVariable","value":"breakfast"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":" "},
  {"type":"CommentSingle","value":"// \"beignets\".\n"},
  {"type":"Text","value":"\n"},
  {"type":"Keyword","value":"if"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"("},
  {"type":"NameVariable","value":"condition"},
  {"type":"Punctuation","value":")"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"Text","value":"\n  "},
  {"type":"Keyword","value":"print"},
  {"type":"Text","value":" "},
  {"type":"LiteralStringDouble","value":"\"yes\""},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n"},
  {"type":"Punctuation","value":"}"},
  {"type":"Text","value":" "},
  {"type":"Keyword","value":"else"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"Text","value":"\n  "},
  {"type":"Keyword","value":"print"},
  {"type":"Text","value":" "},
  {"type":"LiteralStringDouble","value":"\"no\""},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n"},
  {"type":"Punctuation","value":"}"},
  {"type":"Text","value":"\n\n"},
  {"type":"KeywordDeclaration","value":"var "},
  {"type":"NameVariable","value":"a"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"="},
  {"type":"Text","value":" "},
  {"type":"LiteralNumberInteger","value":"1"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n"},
  {"type":"Keyword","value":"while"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"("},
  {"type":"NameVariable","value":"a"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"\u003c"},
  {"type":"Text","value":" "},
  {"type":"LiteralNumberInteger","value":"10"},
  {"type":"Punctuation","value":")"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"Text","value":"\n  "},
  {"type":"Keyword","value":"print"},
  {"type":"Text","value":" "},
  {"type":"NameVariable","value":"a"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n  "},
  {"type":"NameVariable","value":"a"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"="},
  {"type":"Text","value":" "},
  {"type":"NameVariable","value":"a"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"+"},
  {"type":"Text","value":" "},
  {"type":"LiteralNumberInteger","value":"1"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n"},
  {"type":"Punctuation","value":"}"},
  {"type":"Text","value":"\n\n"},
  {"type":"Keyword","value":"for"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"("},
  {"type":"KeywordDeclaration","value":"var "},
  {"type":"NameVariable","value":"a"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"="},
  {"type":"Text","value":" "},
  {"type":"LiteralNumberInteger","value":"1"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":" "},
  {"type":"NameVariable","value":"a"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"\u003c"},
  {"type":"Text","value":" "},
  {"type":"LiteralNumberInteger","value":"10"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":" "},
  {"type":"NameVariable","value":"a"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"="},
  {"type":"Text","value":" "},
  {"type":"NameVariable","value":"a"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"+"},
  {"type":"Text","value":" "},
  {"type":"LiteralNumberInteger","value":"1"},
  {"type":"Punctuation","value":")"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"Text","value":"\n  "},
  {"type":"Keyword","value":"print"},
  {"type":"Text","value":" "},
  {"type":"NameVariable","value":"a"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n"},
  {"type":"Punctuation","value":"}"},
  {"type":"Text","value":"\n\n"},
  {"type":"NameVariable","value":"makeBreakfast"},
  {"type":"Punctuation","value":"("},
  {"type":"NameVariable","value":"bacon"},
  {"type":"Punctuation","value":","},
  {"type":"Text","value":" "},
  {"type":"NameVariable","value":"eggs"},
  {"type":"Punctuation","value":","},
  {"type":"Text","value":" "},
  {"type":"NameVariable","value":"toast"},
  {"type":"Punctuation","value":");"},
  {"type":"Text","value":"\n\n"},
  {"type":"NameVariable","value":"makeBreakfast"},
  {"type":"Punctuation","value":"();"},
  {"type":"Text","value":"\n\n"},
  {"type":"KeywordDeclaration","value":"fun "},
  {"type":"NameFunction","value":"addPair"},
  {"type":"Punctuation","value":"("},
  {"type":"NameVariable","value":"a"},
  {"type":"Punctuation","value":","},
  {"type":"Text","value":" "},
  {"type":"NameVariable","value":"b"},
  {"type":"Punctuation","value":")"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"Text","value":"\n  "},
  {"type":"Keyword","value":"return"},
  {"type":"Text","value":" "},
  {"type":"NameVariable","value":"a"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"+"},
  {"type":"Text","value":" "},
  {"type":"NameVariable","value":"b"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n"},
  {"type":"Punctuation","value":"}"},
  {"type":"Text","value":"\n"},
  {"type":"KeywordDeclaration","value":"fun "},
  {"type":"NameFunction","value":"identity"},
  {"type":"Punctuation","value":"("},
  {"type":"NameVariable","value":"a"},
  {"type":"Punctuation","value":")"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"Text","value":"\n  "},
  {"type":"Keyword","value":"return"},
  {"type":"Text","value":" "},
  {"type":"NameVariable","value":"a"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n"},
  {"type":"Punctuation","value":"}"},
  {"type":"Text","value":"\n"},
  {"type":"Keyword","value":"print"},
  {"type":"Text","value":" "},
  {"type":"NameVariable","value":"identity"},
  {"type":"Punctuation","value":"("},
  {"type":"NameVariable","value":"addPair"},
  {"type":"Punctuation","value":")("},
  {"type":"LiteralNumberInteger","value":"1"},
  {"type":"Punctuation","value":","},
  {"type":"Text","value":" "},
  {"type":"LiteralNumberInteger","value":"2"},
  {"type":"Punctuation","value":");"},
  {"type":"Text","value":" "},
  {"type":"CommentSingle","value":"// Prints \"3\".\n"},
  {"type":"Text","value":"\n"},
  {"type":"KeywordDeclaration","value":"fun "},
  {"type":"NameFunction","value":"returnFunction"},
  {"type":"Punctuation","value":"()"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"Text","value":"\n  "},
  {"type":"KeywordDeclaration","value":"var "},
  {"type":"NameVariable","value":"outside"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"="},
  {"type":"Text","value":" "},
  {"type":"LiteralStringDouble","value":"\"outside\""},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n  "},
  {"type":"KeywordDeclaration","value":"fun "},
  {"type":"NameFunction","value":"inner"},
  {"type":"Punctuation","value":"()"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"Text","value":"\n    "},
  {"type":"Keyword","value":"print"},
  {"type":"Text","value":" "},
  {"type":"NameVariable","value":"outside"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n  "},
  {"type":"Punctuation","value":"}"},
  {"type":"Text","value":"\n  "},
  {"type":"Keyword","value":"return"},
  {"type":"Text","value":" "},
  {"type":"NameVariable","value":"inner"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n"},
  {"type":"Punctuation","value":"}"},
  {"type":"Text","value":"\n"},
  {"type":"KeywordDeclaration","value":"var "},
  {"type":"NameVariable","value":"fn"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"="},
  {"type":"Text","value":" "},
  {"type":"NameVariable","value":"returnFunction"},
  {"type":"Punctuation","value":"();"},
  {"type":"Text","value":"\n"},
  {"type":"NameVariable","value":"fn"},
  {"type":"Punctuation","value":"();"},
  {"type":"Text","value":"\n\n"},
  {"type":"KeywordDeclaration","value":"class "},
  {"type":"NameClass","value":"Breakfast"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"Text","value":"\n  "},
  {"type":"NameVariable","value":"cook"},
  {"type":"Punctuation","value":"()"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"Text","value":"\n    "},
  {"type":"Keyword","value":"print"},
  {"type":"Text","value":" "},
  {"type":"LiteralStringDouble","value":"\"Eggs a-fryin'!\""},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n  "},
  {"type":"Punctuation","value":"}"},
  {"type":"Text","value":"\n  "},
  {"type":"NameVariable","value":"serve"},
  {"type":"Punctuation","value":"("},
  {"type":"NameVariable","value":"who"},
  {"type":"Punctuation","value":")"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"Text","value":"\n    "},
  {"type":"Keyword","value":"print"},
  {"type":"Text","value":" "},
  {"type":"LiteralStringDouble","value":"\"Enjoy your breakfast, \""},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"+"},
  {"type":"Text","value":" "},
  {"type":"NameVariable","value":"who"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"+"},
  {"type":"Text","value":" "},
  {"type":"LiteralStringDouble","value":"\".\""},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n  "},
  {"type":"Punctuation","value":"}"},
  {"type":"Text","value":"\n"},
  {"type":"Punctuation","value":"}"},
  {"type":"Text","value":"\n\n"},
  {"type":"CommentSingle","value":"// Store it in variables.\n"},
  {"type":"KeywordDeclaration","value":"var "},
  {"type":"NameVariable","value":"someVariable"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"="},
  {"type":"Text","value":" "},
  {"type":"NameVariable","value":"Breakfast"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n\n"},
  {"type":"CommentSingle","value":"// Pass it to functions.\n"},
  {"type":"NameVariable","value":"someFunction"},
  {"type":"Punctuation","value":"("},
  {"type":"NameVariable","value":"Breakfast"},
  {"type":"Punctuation","value":");"},
  {"type":"Text","value":"\n\n"},
  {"type":"NameVariable","value":"breakfast"},
  {"type":"Punctuation","value":"."},
  {"type":"NameVariable","value":"meat"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"="},
  {"type":"Text","value":" "},
  {"type":"LiteralStringDouble","value":"\"sausage\""},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n"},
  {"type":"NameVariable","value":"breakfast"},
  {"type":"Punctuation","value":"."},
  {"type":"NameVariable","value":"bread"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"="},
  {"type":"Text","value":" "},
  {"type":"LiteralStringDouble","value":"\"sourdough\""},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n\n"},
  {"type":"KeywordDeclaration","value":"class "},
  {"type":"NameClass","value":"Breakfast"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"Text","value":"\n  "},
  {"type":"NameVariable","value":"init"},
  {"type":"Punctuation","value":"("},
  {"type":"NameVariable","value":"meat"},
  {"type":"Punctuation","value":","},
  {"type":"Text","value":" "},
  {"type":"NameVariable","value":"bread"},
  {"type":"Punctuation","value":")"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"Text","value":"\n    "},
  {"type":"Keyword","value":"this"},
  {"type":"Punctuation","value":"."},
  {"type":"NameVariable","value":"meat"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"="},
  {"type":"Text","value":" "},
  {"type":"NameVariable","value":"meat"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n    "},
  {"type":"Keyword","value":"this"},
  {"type":"Punctuation","value":"."},
  {"type":"NameVariable","value":"bread"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"="},
  {"type":"Text","value":" "},
  {"type":"NameVariable","value":"bread"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n  "},
  {"type":"Punctuation","value":"}"},
  {"type":"Text","value":"\n  "},
  {"type":"CommentSingle","value":"// ...\n"},
  {"type":"Punctuation","value":"}"},
  {"type":"Text","value":"\n"},
  {"type":"KeywordDeclaration","value":"var "},
  {"type":"NameVariable","value":"baconAndToast"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"="},
  {"type":"Text","value":" "},
  {"type":"NameVariable","value":"Breakfast"},
  {"type":"Punctuation","value":"("},
  {"type":"LiteralStringDouble","value":"\"bacon\""},
  {"type":"Punctuation","value":","},
  {"type":"Text","value":" "},
  {"type":"LiteralStringDouble","value":"\"toast\""},
  {"type":"Punctuation","value":");"},
  {"type":"Text","value":"\n"},
  {"type":"NameVariable","value":"baconAndToast"},
  {"type":"Punctuation","value":"."},
  {"type":"NameVariable","value":"serve"},
  {"type":"Punctuation","value":"("},
  {"type":"LiteralStringDouble","value":"\"Dear Reader\""},
  {"type":"Punctuation","value":");"},
  {"type":"Text","value":"\n"},
  {"type":"CommentSingle","value":"// \"Enjoy your bacon and toast, Dear Reader.\"\n"},
  {"type":"Text","value":"\n"},
  {"type":"KeywordDeclaration","value":"class "},
  {"type":"NameClass","value":"Brunch"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"\u003c"},
  {"type":"Text","value":" "},
  {"type":"NameVariable","value":"Breakfast"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"Text","value":"\n  "},
  {"type":"NameVariable","value":"init"},
  {"type":"Punctuation","value":"("},
  {"type":"NameVariable","value":"meat"},
  {"type":"Punctuation","value":","},
  {"type":"Text","value":" "},
  {"type":"NameVariable","value":"bread"},
  {"type":"Punctuation","value":","},
  {"type":"Text","value":" "},
  {"type":"NameVariable","value":"drink"},
  {"type":"Punctuation","value":")"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"Text","value":"\n    "},
  {"type":"Keyword","value":"super"},
  {"type":"Punctuation","value":"."},
  {"type":"NameVariable","value":"init"},
  {"type":"Punctuation","value":"("},
  {"type":"NameVariable","value":"meat"},
  {"type":"Punctuation","value":","},
  {"type":"Text","value":" "},
  {"type":"NameVariable","value":"bread"},
  {"type":"Punctuation","value":");"},
  {"type":"Text","value":"\n    "},
  {"type":"Keyword","value":"this"},
  {"type":"Punctuation","value":"."},
  {"type":"NameVariable","value":"drink"},
  {"type":"Text","value":" "},
  {"type":"Operator","value":"="},
  {"type":"Text","value":" "},
  {"type":"NameVariable","value":"drink"},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n  "},
  {"type":"Punctuation","value":"}"},
  {"type":"Text","value":"\n  "},
  {"type":"NameVariable","value":"drink"},
  {"type":"Punctuation","value":"()"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"Text","value":"\n    "},
  {"type":"Keyword","value":"print"},
  {"type":"Text","value":" "},
  {"type":"LiteralStringDouble","value":"\"How about a Bloody Mary?\""},
  {"type":"Punctuation","value":";"},
  {"type":"Text","value":"\n  "},
  {"type":"Punctuation","value":"}"},
  {"type":"Text","value":"\n"},
  {"type":"Punctuation","value":"}"},
  {"type":"Text","value":"\n"}
]
